### PR TITLE
8315241: (fs) Move toRealPath tests in java/nio/file/Path/Misc.java to separate JUnit 5 test

### PR DESCRIPTION
**Pull request by bplb**
Move toRealPath() tests from Misc.java to ToRealPath.java.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315241](https://bugs.openjdk.org/browse/JDK-8315241): (fs) Move toRealPath tests in java/nio/file/Path/Misc.java to separate JUnit 5 test (**Enhancement** - P4)

### Reviewers
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**) ⚠️ Review applies to [b3547aeb](https://git.openjdk.org/jdk/pull/15474/files/b3547aeb8a6553e4fd40acb4513ce219b5eb2be2)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15474/head:pull/15474` \
`$ git checkout pull/15474`

Update a local copy of the PR: \
`$ git checkout pull/15474` \
`$ git pull https://git.openjdk.org/jdk.git pull/15474/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15474`

View PR using the GUI difftool: \
`$ git pr show -t 15474`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15474.diff">https://git.openjdk.org/jdk/pull/15474.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15474#issuecomment-1697927516)

### CODE REVIEW

**Review by RogerRiggs — APPROVED** [review]
lgtm

File: test/jdk/java/nio/file/Path/ToRealPath.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+public class ToRealPath {
+    static final boolean SUPPORTS_LINKS;
+    static final Path DIR;
+    static final Path SUBDIR;
+    static final Path FILE;
+    static final Path LINK;
+
+    static {
+        try {
+            DIR  = TestUtil.createTemporaryDirectory();
```
Comment (turbanoff) [inline test/jdk/java/nio/file/Path/ToRealPath.java:57]:
```suggestion
            DIR = TestUtil.createTemporaryDirectory();
```

Comment (bplb) [reply test/jdk/java/nio/file/Path/ToRealPath.java]: Fixed in 34511d8814856f6eedbf742ed959533f3592bebb, thanks.

**Pull request closed without merging**